wdl/src/test/scala/wdl/expression/PureStandardLibraryFunctionsSpec.scala

```diff
@@ -36,6 +36,35 @@ class PureStandardLibraryFunctionsSpec extends FlatSpec with Matchers {
     PureStandardLibraryFunctions.length(Seq(Success(empty))) should be(Success(WomInteger(0)))
   }
 
+  behavior of "flatten"
+
+  it should "get the right answers" in {
+    val ar1 = WomArray(WomArrayType(WomIntegerType), List(WomInteger(1), WomInteger(2)))
+    val ar2 = WomArray(WomArrayType(WomIntegerType), List(WomInteger(3), WomInteger(4), WomInteger(5)))
+    val ar3 = WomArray(WomArrayType(WomIntegerType), List.empty)
+    val ar4 = WomArray(WomArrayType(WomIntegerType), List(WomInteger(6)))
+    val aar = WomArray(WomArrayType(WomArrayType(WomIntegerType)), List(ar1, ar2, ar3, ar4))
+    val flat_ar = WomArray(WomArrayType(WomIntegerType), List(WomInteger(1), WomInteger(2),
+                                                              WomInteger(3), WomInteger(4),
+                                                              WomInteger(5), WomInteger(6)))
+    PureStandardLibraryFunctions.flatten(Seq(Success(aar))) should be(Success(flat_ar))
+
+    val sar1 = WomArray(WomArrayType(WomStringType), List(WomString("chatting"), WomString("is")))
+    val sar2 = WomArray(WomArrayType(WomStringType), List(WomString("great"), WomString("for"), WomString("you")))
+    val saar = WomArray(WomArrayType(WomArrayType(WomStringType)), List(sar1, sar2))
+    val flat_sar = WomArray(WomArrayType(WomStringType), List(WomString("chatting"), WomString("is"),
+                                                              WomString("great"), WomString("for"),
+                                                              WomString("you")))
+    PureStandardLibraryFunctions.flatten(Seq(Success(saar))) should be(Success(flat_sar))
+
+    val err1 = WomArray(WomArrayType(WomFileType), List.empty)
```

Comment:
In a world where `flatten` supports flattening deeper nestings than 2, I'm not sure I particularly object to being able to flatten already-flat arrays either (eg calling flatten on a flat array is just the identity function)